### PR TITLE
session: fix log slow log succ field

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
The `succ` field in the slow log was wrong when logging slow log for DDL/DML and `admin show slow`.

### Check List 

Tests 

 - manual test.

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch(affect version: v2.1.16, v3.0.2)

Release note

 - Fix the issue that `succ` field in slow log and `admin show low` was wrong.

### CODE REVIEW

**Comment by crazycs520** [comment]
@jackysp @zz-jason PTAL

File: session/tidb.go [hunk 1]
```diff
@@ -214,7 +214,7 @@ func runStmt(ctx context.Context, sctx sessionctx.Context, s sqlexec.Statement)
 		// If it is not a select statement, we record its slow log here,
 		// then it could include the transaction commit time.
 		if rs == nil {
-			s.(*executor.ExecStmt).LogSlowQuery(se.GetSessionVars().TxnCtx.StartTS, err != nil)
+			s.(*executor.ExecStmt).LogSlowQuery(se.GetSessionVars().TxnCtx.StartTS, err == nil)
```
Comment (zz-jason) [inline session/tidb.go]: please add a UT.

Comment (crazycs520) [reply session/tidb.go]: It's hard to add UT to test this..., any suggestion？

Comment (zz-jason) [reply session/tidb.go]: how about duplicated primary key error?

Comment (crazycs520) [reply session/tidb.go]: ? To test this problem, we need the `slow log` output and  get the SQL corresponding slow log, then check the `succ` field. 

The above method is hard to implement in UT.

**Pull request merged**